otello/base.py

```diff
@@ -12,36 +12,49 @@ def __init__(self, cfg=None, session=None):
             cfg = os.path.join(cfg_dir, 'config.yml')
 
-        self._cfg_file = cfg
-        try:
+        if isinstance(cfg, str):
+            self._cfg_file = cfg
             with open(cfg, 'r') as f:
                 self._cfg = yaml.safe_load(f)
-        except FileNotFoundError as e:
-            raise FileNotFoundError(e)
-        except yaml.YAMLError as e:
-            raise yaml.YAMLError(e)
-        except Exception as e:
-            raise Exception(e)
+                cfg_loaded_from_file = True
+        elif isinstance(cfg, dict):
+            self._cfg = cfg
+        else:
+            raise TypeError("cfg must be a path to a yaml file or a dict")
 
         if session:
             self._session = session
         else:
             self._session = requests.Session()
+
             if self._cfg["auth"] is True:
-                try:
-                    client = boto3.client("secretsmanager")
-                    response = client.get_secret_value(
-                        SecretId=self._cfg["aws_secret_id"]
-                    )
-                    secret_string = json.loads(response["SecretString"])
+                if self._cfg["username"] is None:
+                    raise ValueError("No username provided")
+
+                if self._cfg["aws_secret_id"] is not None:
+                    try:
+                        client = boto3.client("secretsmanager")
+                        response = client.get_secret_value(
+                            SecretId=self._cfg["aws_secret_id"]
+                        )
+                        secret_string = json.loads(response["SecretString"])
+                        self._session.auth = (self._cfg["username"],
+                                              secret_string[self._cfg["username"]])
+                    except Exception as e:
+                        raise Exception(f"Error occurred while trying to set "
+                                        f"authentication using AWS Secrets "
+                                        f"Manager:\n{str(e)}")
+                elif self._cfg["password"] is not None:
+                    if cfg_loaded_from_file:
+                        raise ValueError("Password provided in a plaintext "
+                                         "file. Please remove password from "
+                                         "config.yml and use AWS Secrets Manager "
+                                         "instead")
+
                     self._session.auth = (self._cfg["username"],
-                                          secret_string[self._cfg["username"]]
-                                          )
-                except Exception as e:
-                    raise Exception(f"Error occurred while trying to set "
-                                    f"authentication using AWS Secrets "
-                                    f"Manager:\n{str(e)}")
-            self._session.verify = False
```

Comment:
I recall during testing that if this was set to True, that our authentication would end up failing. I'm wondering if this should be set to False when Otello needs to create the session object. Either that or perhaps we could wire this out and let the caller determine what to set this to?

Reply:
```py
self._session.verify = False
```
This actually a security issue because it disables SSL certificate checking globally throughout Ortello. The proper fix should be to have whatever system is running Otello add the certificate CAs it's missing to the system's certificate store (in my case, it was the JPL CA missing) and set the `REQUESTS_CA_BUNDLE` environment variable to have requests utilize the system's CA store.

https://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification

Reply:
Ya I totally understand. I think the issue is that the `verify=False` setting has always been here and was just moved to the init step due to: https://github.com/hysds/otello/pull/11

If we change this to True, then this will likely break things on the NISAR side, where Otello is deployed already and configured on working systems. I think in general, we should make things backwards compatible, but I'm open here. 
@pymonger and @DustinKLo could you weigh in on this?

Reply:
I really don't think that keeping a security bug unpatched for backwards compatibility is the best move here; especially Otello would be sending out cleartext of people's usernames and passwords since it's vulnerable to man in the middle attacks.

Reply:
we can add `verify` as a parameter in the constructor and default to `True/False`
forcing `verify=True` will break current usage in other projects

Reply:
I would not advise actively supporting this use case directly. If a user wants to disable SSL verification, Otello provides a way to directly inject a requests.Session object which can be configured to disable SSL verification. It should be left as an advanced, configurable feature outside the code base.

Reply:
If we make it configurable, can the default be set to `True` at least? You'd have to update NISAR to explicitly set it to `False` to keep the current behavior; but it would prevent new users in the future from disabling SSL verification by default.

Reply:
Alternatively, we could propose this as a MAJOR version increment of otello to indicate the backwards-incompatible change; other clients could choose to upgrade or not.

Reply:
Okay would a compromise be that we have a 1.1.0 (current 1.0.2) which maintains backwards compatibility and adds that depreciation warning and then bump to 2.0.0 to default to True like Frank suggested?

We can break this PR out then into just the API configuration changes and open a separate PR for 2.0.0

Reply:
Sounds like a plan. @pymonger you ok with this too?

Reply:
@mcayanan @pymonger @DustinKLo The changes requested have been added. Is there anything else that needs to be updated?

Reply:
Thanks for making these recent changes. @DustinKLo and/or @pymonger any concerns from a NISAR Jupyter notebook point of view?

Reply:
sounds good to me too. thanks @joshgarde.